sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCVCIssuancePlugin.java

```diff
@@ -0,0 +1,260 @@
+    public static final String STATIC_VALUE_MAP_ISSUER_ID = "static-value-map.issuerId";
+
+    @Autowired
+    Environment env;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${mosip.esignet.vciplugin.sunbird-rc.issue-credential-url}")
+    String issueCredentialUri;
+
+    @Value("${mosip.esignet.vciplugin.sunbird-rc.supported-credential-types}")
+    String supportedCredentialTypes;
+
+    @Value("${mosip.esignet.vciplugin.sunbird-rc.credential-type.InsuranceCredential.registry-get-url}")
+    String registryUrl;
+
+
+
+    private final Map<String, Template> credentialTypeTemplates = new HashMap<>();
+
+    private final Map<String,Map<String,String>> credentialTypeConfigMap = new HashMap<>();
+
+    private VelocityEngine vEngine;
+
+
+    @PostConstruct
+    public  void validateProperties() throws VCIExchangeException {
+
+        vEngine = new VelocityEngine();
+        vEngine.setProperty(RuntimeConstants.RESOURCE_LOADER, "classpath");
+        vEngine.setProperty("classpath.resource.loader.class", ClasspathResourceLoader.class.getName());
+        vEngine.init();
+        //Validate all the supported VC
+        String[] credentialTypes = supportedCredentialTypes.split(",");
+        for (String credentialType : credentialTypes) {
+            validatePropertyForCredentialType(credentialType.trim());
+        }
+    }
+
+    @Override
+    public VCResult<JsonLDObject> getVerifiableCredentialWithLinkedDataProof(VCRequestDto vcRequestDto, String holderId, Map<String, Object> identityDetails) throws VCIExchangeException {
+        if (vcRequestDto == null || vcRequestDto.getType() == null) {
+            throw new VCIExchangeException(ErrorConstants.VCI_EXCHANGE_FAILED);
+        }
+        List<String> types = vcRequestDto.getType();
+        if (types.isEmpty() || !types.get(0).equals("VerifiableCredential")) {
+            log.error("Invalid request: first item in type is not VerifiableCredential");
+            throw new VCIExchangeException(ErrorConstants.VCI_EXCHANGE_FAILED);
+        }
+        types.remove(0);
+        String requestedCredentialType = String.join("-", types);
+        //Check if the key is in the supported-credential-types
+        List<String> supportedTypes = Arrays.asList(supportedCredentialTypes.split(","));
+        if (!supportedTypes.contains(requestedCredentialType)) {
+            log.error("Credential type is not supported");
+            throw new VCIExchangeException(ErrorConstants.VCI_EXCHANGE_FAILED);
+        }
+        //todo validate context of vcrequestdto with template
+        List<String> contextList=vcRequestDto.getContext();
+        for(String supportedType:supportedTypes){
+            Template template=credentialTypeTemplates.get(supportedType);
+            validateContextUrl(template,contextList);
+        }
+        String osid=(String)identityDetails.get("sub");
+        RequestEntity requestEntity = RequestEntity
```

Comment:
This method looks too long, can you rather create 2 new methods one for fetchRegistryObject and another for sendCredentialIssueRequest. So you can call the methods in below sequence from this method. 
1. fetchRegistryObject
2. createCredentialIssueRequest
3. sendCredentialIssueRequest

Finaly construct the vcResult object here and return

Exception throwing, let it happen in the respective above methods

Reply:
splitted